commonmark/src/main/java/org/commonmark/node/AutoLink.java

```diff
@@ -0,0 +1,17 @@
+package org.commonmark.node;
+
+public class AutoLink extends Link {
```

Comment:
It makes sense that `AutoLink` is a separate node type.

I'm not sure about subclassing `Link` though. After all, it doesn't have all the attributes that a `Link` has (no label or title). Have you thought about having it extend `Node` instead?